lib/routes/zhihu/all-collections.ts

```diff
@@ -0,0 +1,114 @@
+
+    const response = await got(String(apiPath), {
+        headers: {
+            Referer: `https://www.zhihu.com/people/${id}/collections`,
+        },
+    });
+
+    const collections = response.data.data;
+
+    const allCollectionItems = await Promise.all(
+        collections.map(async (collection) => {
+            const firstPageResponse = await got({
+                method: 'get',
+                url: `https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=0&limit=20`,
+                headers: {
+                    ...header,
+                    Referer: `https://www.zhihu.com/collection/${collection.id}`,
+                },
+            });
+
+            const {
+                data: items,
+                paging: { totals },
+            } = firstPageResponse.data;
+
+            if (totals > 20) {
+                const offsetList = Array.from({ length: Math.ceil(totals / 20) - 1 }, (_, index) => (index + 1) * 20);
+
+                const otherPages = await Promise.all(
+                    offsetList.map((offset) =>
+                        cache.tryGet(`https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=${offset}&limit=20`, async () => {
+                            const response = await got({
+                                method: 'get',
+                                url: `https://www.zhihu.com/api/v4/collections/${collection.id}/items?offset=${offset}&limit=20`,
+                                headers: {
+                                    ...header,
+                                    Referer: `https://www.zhihu.com/collection/${collection.id}`,
+                                },
+                            });
+                            return response.data.data;
+                        })
+                    )
+                );
+
+                items.push(...otherPages.flat());
+            }
+
+            return {
+                collectionId: collection.id,
+                collectionTitle: collection.title,
+                items,
+            };
+        })
+    );
+
+    const items = allCollectionItems.flatMap((collection) =>
+        collection.items.map((item) => ({
+            ...item,
+            collectionTitle: collection.collectionTitle,
+        }))
+    );
+
+    return {
+        title: `${collections[0].creator.name}的知乎收藏`,
+        link: `https://www.zhihu.com/people/${id}/collections`,
+        item: items.map((item) => {
+            const content = item.content;
+
+            const transformed = {
+                title: content.type === 'article' || content.type === 'zvideo' ? content.title : content.question.title,
+                link: content.url,
+                description: content.type === 'zvideo' ? `<img src=${content.video.url}/>` : content.content,
+                pubDate: parseDate((content.type === 'article' ? content.updated : content.updated_time) * 1000),
+                collectionTitle: item.collectionTitle,
```

Comment:
This property is not listed in https://github.com/DIYgod/RSSHub/blob/55b6afeed45e910e5fedfd0d5c0750df89572be7/lib/types.ts#L32